lib/src/default_index/store.rs

```diff
@@ -316,10 +320,11 @@ impl DefaultIndexStore {
         );
         // Build a list of ancestors of heads where parents come after the
         // commit itself.
-        let parent_index_has_id = |id: &CommitId| {
-            maybe_parent_index
-                .as_ref()
-                .is_some_and(|index| index.has_id(id))
+        let parent_index_has_id = |id: &CommitId| -> IndexResult<bool> {
+            match maybe_parent_index.as_ref() {
+                Some(parent_index) => parent_index.has_id(id),
```

Comment:
I don't know if it will be simpler overall but I suppose we could call a separate `parent_index.has_id_impl()` that returns a plain `bool` here so we don't have to handle errors. Try and see if it's simpler and just let me know if it seems worse?

Reply:
I'll take a look at this. Thanks!

Reply:
Just made this update -- I think it's much better now. I took your `has_id_impl` naming suggestion literally -- please LMK if you'd prefer another name.